### PR TITLE
MSPT-1917 - lsp-legid-setter fail queue

### PR DESCRIPTION
**Pull request by gpuicentric**
Raising WIP MR for early review.

### CODE REVIEW

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java [hunk 1]
```diff
@@ -1,26 +1,62 @@
...
+        IspyContext ispyContext = LambdaEventIspyContext.getIspyContextFromEvent(event);
+
+        Optional<LivestreamRecord> livestreamRecordOptional = lspMedialiveStateClient.getLivestreamRecord(cvid);
+
+        Optional<ChannelRecord> channelRecordOptional = lspMedialiveStateClient.getChannelRecord(cvid);
+        final ChannelRecord channelRecord = channelRecordOptional
+                .orElseThrow(() -> new StateApiException(String.format("Channel record not found for cvid: %s", cvid), 200));
+
+
+        if (!SIMULCAST.equalsIgnoreCase(livestreamRecordOptional.get().getWorkflowType())) {
```
Comment (xuq01) [inline src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:49]: need to check livestreamRecordOptional isPresent

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java [hunk 2]
```diff
@@ -1,26 +1,62 @@
...
+
+            if (legId.equals(channelRecord.getLegId())) {
+                ispyContext.ispy("info.identical-leg-id-exists");
```
Comment (xuq01) [inline src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:52]: Why the ispy event is different from the ticket?

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java [hunk 3]
```diff
@@ -1,26 +1,62 @@
...
+            } else {
+                ispyContext.ispy("error.different-leg-id-exists.notification");
```
Comment (xuq01) [inline src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:54]: Why the ispy event is different from the ticket?

File: component-test/cucumber/features/lsp-legid-setter.feature [hunk 4]
```diff
@@ -3,19 +3,42 @@ Feature:
...
+
+  Scenario: Get stream details from Ribbon and when different leg-id exist and the workflow type is not simulcast
+    Given lsp-legid-setter lambda exists
+    And Ribbon Get leg endpoint will respond with 200
+    And Medialive State Api Get live stream endpoint will respond with 200 and workflow type "not simulcast"
+    And Medialive State Api Get channels endpoint will respond with 200 and leg id "duplicateLegId"
+    When a Livestream Created message is sent to the queue
+    Then the iSpy event "lsp-legid-setter.livestream-created.received" is emitted
+    And Ribbon Get leg endpoint is called
+    And the iSpy event "lsp-legid-setter.error.different-leg-id-exists.notification" is emitted
```
Comment (xuq01) [inline component-test/cucumber/features/lsp-legid-setter.feature:34]: I am not sure how these 2 scenarios match the ones in the ticket, but one of them should check message appear in fail queue

**Review by xuq01 — COMMENTED** [review]
It looks good. There are some issues need to fix. And the ispy events are different from the ones in ticket

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java (hunk 3 above)
Comment (gpuicentric) [reply src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:54]: Just updated the ticket to match, after checking with Kevin Morris

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java (hunk 1 above)
Comment (gpuicentric) [reply src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:49]: Just updated the ticket to match, after checking with Kevin Morris

File: src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java [hunk 5]
```diff
@@ -45,15 +45,17 @@ public LambdaEvent<SQSMessage> apply(LambdaEvent<SQSMessage> event) throws Excep
         final ChannelRecord channelRecord = channelRecordOptional
                 .orElseThrow(() -> new StateApiException(String.format("Channel record not found for cvid: %s", cvid), 200));
 
+        if (livestreamRecordOptional.isPresent()) {
```
Comment (xuq01) [inline src/main/java/uk/co/bbc/lsp_legid_setter/handler/GetLegIdFromRibbon.java:48]: Sorry what I mean is to throw out exception like line 46 if it is not present

**Review by xuq01 — COMMENTED** [review]
only a small issue need to fix

**Pull request merged**